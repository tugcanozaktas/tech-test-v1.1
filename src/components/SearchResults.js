import React from "react";
import "../styles/SearchResults.css"

const SearchResults = ({searchResults}) =>{

  return (
    <>
      <p><b><i>Search Results</i></b></p>
      <div className="search-results">
        {searchResults.map((link, index) => (
          <img key={index} className="image-card"  src={link} alt="spaceImage" />))}
        
      </div>
    </>
  )
};

export default SearchResults;