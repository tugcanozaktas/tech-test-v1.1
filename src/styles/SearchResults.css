.search-results{
    margin-top: 50px;
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    gap: 50px;
    margin-bottom: 50px;
}
.image-card{
    width: 400px;
    height: 300px;
    border-radius: 10px;
}
@media screen and (max-width: 800px) {
    .image-card{
        width: 250px;
        height: 150px;
    }
}