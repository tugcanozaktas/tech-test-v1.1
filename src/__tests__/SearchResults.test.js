import React from 'react';
import { render } from '@testing-library/react';
import SearchResults from '../components/SearchResults.js';

describe("SearchResults", () => {
    const search = ["https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"]

    const { asFragment } = render(<SearchResults searchResults={search} />);

    it("renders correctly", () =>{
        expect(asFragment()).toMatchSnapshot();
    })
})