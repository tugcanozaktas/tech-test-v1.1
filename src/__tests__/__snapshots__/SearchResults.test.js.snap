// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p>
    <b>
      <i>
        Search Results
      </i>
    </b>
  </p>
  <div
    class="search-results"
  >
    <img
      alt="spaceImage"
      class="image-card"
      src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
    />
  </div>
</DocumentFragment>
`;
